src/ui/confwin.c

```diff
@@ -340,14 +340,8 @@ confwin_get_string(ProfConfWin* confwin)
 {
     assert(confwin != NULL);
 
-    GString* res = g_string_new("");
-
     char* title = win_get_title((ProfWin*)confwin);
-    g_string_append(res, title);
+    char* result = g_strdup_printf("%s", title);
     free(title);
-
-    char* resstr = res->str;
-    g_string_free(res, FALSE);
-
-    return resstr;
+    return result;
```

Comment:
```suggestion
    return win_get_title((ProfWin*)confwin);
```
Why first dup the result or `win_get_title()`, then free the source and return the dup? Simply return what is returned by `win_get_title()` or not?

Also I've just realized that `win_get_title()` can return `NULL`, maybe make it return an empty string on error?

Reply:
The reason why I mention it is that the return value of `win_get_title()` is possibly `NULL` and will directly be used e.g. here

https://github.com/profanity-im/profanity/blob/e5e8ff221a08939b43edf488fa2a3b8fe95169ea/src/ui/titlebar.c#L198-L201

1. How does `mvwprintw()` handle a potential `NULL` pointer?
2. I'm pretty sure that most `strlen()` implementations will not take it well :grimacing: 

Just to make it clear: this isn't caused by this change, but was already before like that. I only discovered it while looking at your PR.

I couldn't find a case where the `NULL` was really tested for to decide on behavior, so maybe the best solution would be to also return `strdup(CONS_WIN_TITLE)` in the fall-through case? or maybe even `g_strdup_printf("Error: Window title for type %d unknown", window->type)` to make it clear that something went wrong?